src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotFirecloudGroupNameStep.java

```diff
@@ -0,0 +1,40 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.auth.iam.exception.IamNotFoundException;
+import bio.terra.service.duos.DuosService;
+import bio.terra.service.snapshot.flight.SnapshotWorkingMapKeys;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.UUID;
+
+public class CreateSnapshotFirecloudGroupNameStep implements Step {
+  private final UUID snapshotId;
+  private final IamService iamService;
+
+  public CreateSnapshotFirecloudGroupNameStep(UUID snapshotId, IamService iamService) {
+    this.snapshotId = snapshotId;
+    this.iamService = iamService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    String groupName = DuosService.constructUniqueFirecloudGroupName(String.valueOf(snapshotId));
```

Comment:
Would it make sense to move this method to a common location, like `IamService` or `SamIam`?

Also, I believe the term `firecloud group` is a DUOS specific term. The Sam API calls it `group`, so I would use the Sam name for this method, not the DUOS name.

Reply:
Yes I'll move them to IamService. And about naming - the steps are called firecloud as well. I didn't think this terminology was duos specific. I originally called the steps create sam group, but on my previous PR it was suggested that I change it to match the terminology used for duos.